sessionctx/variable/session.go

```diff
@@ -820,6 +826,18 @@ type SessionVars struct {
 	AllowFallbackToTiKV map[kv.StoreType]struct{}
 }
 
+// AllocMPPTaskID allocates task id for mpp tasks. It will reset the task id if the query's
+// startTs is different.
+func (s *SessionVars) AllocMPPTaskID(startTS uint64) int64 {
```

Comment:
need a mutex to avoid write conflicts?

Reply:
it's only required by one session. You can notice that every variable in SessionVars is vulnerable.